rules/REQUEST-944-APPLICATION-ATTACK-JAVA.conf

```diff
@@ -142,11 +142,80 @@ SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES
     setvar:'tx.anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
 
+# Log4J / Log4Shell Defense
+#
+# This addresses exploits against the Log4J library described in several CVEs:
+# * CVE-2021-44228
+# * CVE-2021-44832
+# * CVE-2021-45046
+# * CVE-2021-45105
+#
+# See https://coreruleset.org/20211213/crs-and-log4j-log4shell-cve-2021-44228/
+#
+# Rule attempts to detect two things:
+# * Nested use of ${
+# * use of ${jndi:... without the closing bracket
+#
+# Rule 932130 is also essential for defense since there are certain
+# bypasses of the log4j rules that can be caught by 932130.
+#
+# The payload is not displayed in the alert message since log4j could
+# potentially be executed on the logviewer.
+#
+# This rule has a stricter sibling: 944141 (PL3)
+
+SecRule REQUEST_LINE|ARGS|ARGS_NAMES|REQUEST_COOKIES|REQUEST_COOKIES_NAMES|REQUEST_HEADERS|XML:/*|XML://@* "@rx (?:\${[^}]{0,15}\${|\${(?:jndi|ctx))" \
+    "id:944140,\
+    phase:2,\
+    block,\
+    t:none,t:urlDecodeUni,t:cmdLine,\
+    log,\
+    msg:'Potential Remote Command Execution: Log4j / Log4shell', \
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS',\
+    tag:'capec/1000/152/137/6',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/1',\
+    ver:'OWASP_CRS/3.4.0-dev',\
+    severity:'CRITICAL',\
+    setvar:'tx.rce_score=+%{tx.critical_anomaly_score}',\
+    setvar:'tx.anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
+
 SecRule TX:EXECUTING_PARANOIA_LEVEL "@lt 2" "id:944013,phase:1,pass,nolog,skipAfter:END-REQUEST-944-APPLICATION-ATTACK-JAVA"
 SecRule TX:EXECUTING_PARANOIA_LEVEL "@lt 2" "id:944014,phase:2,pass,nolog,skipAfter:END-REQUEST-944-APPLICATION-ATTACK-JAVA"
 #
 # -= Paranoia Level 2 =- (apply only when tx.executing_paranoia_level is sufficiently high: 2 or higher)
 #
+
+# This is a stricter sibling of 944140.
+# It is a re-iteration of said rule without the curly bracket distance limiter
```

Comment:
```suggestion
# It is a re-iteration of said rule without the curly brace distance limiter
```

Reply:
We're not using the term "curly brace" so far, but several uses of synonym "curly bracket". So I'd say we stick to that.

Reply:
Fine by me.